lib/koala/api/graph_batch_api.rb

```diff
@@ -41,67 +41,106 @@ def graph_call_in_batch(path, args = {}, verb = "get", options = {}, &post_proce
 
       # execute the queued batch calls
       def execute(http_options = {})
-        return [] unless batch_calls.length > 0
+        return [] if batch_calls.empty?
+
         # Turn the call args collected into what facebook expects
-        args = {}
-        args["batch"] = JSON.dump(batch_calls.map { |batch_op|
-          args.merge!(batch_op.files) if batch_op.files
-          batch_op.to_batch_params(access_token, app_secret)
-        })
+        args = { 'batch' => batch_args }
+        batch_calls.each do |call|
+          args.merge! call.files || {}
+        end
 
-        batch_result = graph_call_outside_batch('/', args, 'post', http_options) do |response|
-          unless response
-            # Facebook sometimes reportedly returns an empty body at times
-            # see https://github.com/arsduo/koala/issues/184
-            raise BadFacebookResponse.new(200, '', "Facebook returned an empty body")
-          end
+        graph_call_outside_batch('/', args, 'post', http_options, &handle_response)
+      end
+
+      def handle_response
+        lambda do |response|
```

Comment:
I like idea of shortening the method, but splitting up a block into a separate method that returns a 2-line lambda seems overly complicated. Could we merge this back in?